Games/GrayscaleTest/GrayscaleTest.py

```diff
@@ -4,62 +4,146 @@
+# Find where while loops should be for loops
+
+# TODO: test brightness on both GPU and BW
 
 # Helper function
-
 def waitKey():
     while actionPressed(): pass
     while not actionPressed(): pass
     while actionPressed(): pass
 
-# Directly writing to the buffers
+# Drawing primitives demo
 
-for s in range(4):
-    if s & 1:
-        m1 = 0xff
-    else:
-        m1 = 0
-    if s & 2:
-        m2 = 0xff
-    else:
-        m2 = 0
-    sx = s * 18
-    for y in range(5):
-        sy = y * 72
-        for x in range(18):
-            gs.buffer1[sy + sx + x] = m1
-            gs.buffer2[sy + sx + x] = m2
-gs.show()
-waitKey()
-
-# Drawing primitives
-
-gs.drawFilledRectangle(0, 0, 72, 40, gs.WHITE)
-gs.drawFilledRectangle(0, 0, 62, 30, gs.LIGHTGRAY)
-gs.drawFilledRectangle(0, 0, 52, 20, gs.DARKGRAY)
-gs.drawFilledRectangle(0, 0, 42, 10, gs.BLACK)
-gs.drawText("Hello", 2, 31, gs.LIGHTGRAY)
-gs.drawText("world!", 37, 31, gs.DARKGRAY)
+gs.fill(gs.DARKGRAY)
+gs.drawFilledRectangle(16, 9, 40, 21, gs.LIGHTGRAY)
+gs.drawRectangle(14, 7, 44, 25, gs.LIGHTGRAY)
+gs.drawLine(14, 32, 55, 39, gs.LIGHTGRAY)
+gs.drawLine(58, 7, 71, 30, gs.LIGHTGRAY)
+gs.drawText("Hello", 18, 11, gs.WHITE)
+gs.drawText("world!", 18, 19, gs.DARKGRAY)
+gs.update()
+sleep_ms(500)
+
+gs.startGPU()
+
+gs.fill(gs.DARKGRAY)
+gs.drawFilledRectangle(16, 9, 40, 21, gs.LIGHTGRAY)
+gs.drawRectangle(14, 7, 44, 25, gs.LIGHTGRAY)
+gs.drawLine(14, 32, 55, 39, gs.LIGHTGRAY)
+gs.drawLine(58, 7, 71, 30, gs.LIGHTGRAY)
+gs.drawText("Hello", 18, 11, gs.WHITE)
+gs.drawText("world!", 18, 19, gs.DARKGRAY)
 gs.update()
-waitKey()
+sleep_ms(1000)
 
-# Sprites as full screen images
+# Bouncing cat demo
 
-girlSprite = grayscale.Sprite(72, 40, bytearray([
+cat = legacy_sprite(grayscale.ShadedSprite(
+    12, 9,         # Dimensions
+    bytearray([    # Layer 2 data
+        255,255,87,7,3,3,3,67,3,7,7,255,
+        1,1,1,0,0,0,0,0,0,0,1,1
+    ]),
+    bytearray([    # Layer 1 data
+        175,7,169,254,237,255,191,157,190,233,255,175,
+        1,1,0,1,1,1,1,1,1,1,1,1
+    ]),
+    30, 15         # Position
+))
+catMask = grayscale.ShadedSprite(
+    12, 9,         # Dimensions
+    bytearray([0,7,1,0,1,1,1,1,0,1,7,223,
+           1,1,0,0,0,0,0,0,0,0,1,1]),
+    bytearray([0,7,1,0,1,1,1,1,0,1,7,223,
+           1,1,0,0,0,0,0,0,0,0,1,1]),
+    30, 15         # Position
+)
+
+thumby.display = gs
+
+gs.setFPS(60)
+
+while(thumby.buttonA.pressed() == False and thumby.buttonB.pressed() == False):
+    if(ticks_ms() % 1000 < 500):
+        thumby.display.drawFilledRectangle(0, 32, 72, 8, 0)
+        thumby.display.drawText("Press A/B", 9, 32, 1)
+    else:
+        thumby.display.drawFilledRectangle(0, 32, 72, 8, 1)
+        thumby.display.drawText("Press A/B", 9, 32, 0)
+    thumby.display.update()
+    pass
+
+c = dx = dy = 1
+while c < 200:
+    gs.fill(gs.LIGHTGRAY)
+    gs.drawSpriteWithMask(cat, catMask)
+    cat.x += dx
+    cat.y += dy
+    if cat.x == 0 or cat.x == 60:
+        dx = -1 * dx
```

Comment:
I love it. Much tighter! Pushed your suggestion in all the bouncing cat sprite tests! Nice. Note one still exists for on the main branch in case we abandon this branch.